pkg/services/ngalert/schedule/compat.go

```diff
@@ -125,3 +126,19 @@ func FromAlertStateToPostableAlerts(firingStates []*state.State, stateManager *s
 	stateManager.Put(sentAlerts)
 	return alerts
 }
+
+// FromAlertsStateToStoppedAlert converts firingStates that have evaluation state either eval.Alerting or eval.NoData or eval.Error to models.PostableAlert that are accepted by notifiers.
+// Returns a list of alert instances that have expiration time.Now
+func FromAlertsStateToStoppedAlert(firingStates []*state.State, appURL *url.URL, clock clock.Clock) apimodels.PostableAlerts {
```

Comment:
How come we use `clock.Clock` instead of `time.Time`?

Reply:
I use clock because we have tooling to mock it. Same as in scheduler.